File: LeetCode/Study Plan/Practice Problems/Easy/Smallest Range I/Cpp/smallest_range_one.cc
#include <iostream>
#include <vector>

typedef std::vector<int> vi;

int smallest_range_one(vi v, int k) {
    sort(v.begin(), v.end());
    int min_val = v[0] + k;
    int max_val = v[v.size()-1] - k;
    return std::max(0, max_val-min_val);
}

int main() {
    std::cout<<smallest_range_one(vi{1}, 0)<<"\n";
}

File: LeetCode/Easy/Smallest Range I/Go/smallest_range_one.go
package main

import (
	"fmt"
	"sort"
)

func smallest_range_one(arr []int, k int) int {
	sort.Ints(arr)
	min_val := arr[0] + k
	max_val := arr[len(arr)-1] - k
	return max(0, max_val-min_val)
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	fmt.Println(smallest_range_one([]int{1}, 0))
}
